src/blk/BlockDevice.cc

```diff
@@ -226,5 +235,13 @@ void BlockDevice::collect_alerts(osd_alert_list_t& alerts, const std::string& de
       alerts.emplace(device_name + "_DEVICE_STALLED_READ_ALERT", ss.str());
     }
   }
+  if (cct->_conf->bdev_enable_discard && check_discard_queue_overload()) {
+    size_t current_discarded_bytes = discarded_bytes.load();
+    uint64_t current_discard_queue_items = discard_queue_length.load();
+    std::ostringstream ss;
+    ss << "observed " << byte_u_t(current_discarded_bytes) << " bytes and "
+       << current_discard_queue_items << " items in discard queue for " << device_name << " device ";
+    alerts.emplace(device_name + "_DEVICE_DISCARD_QUEUE", ss.str());
```

Comment:
May be pass alerts to check_discard_queue_overflow() and implement all the alert raising logic there?